src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -42,6 +42,8 @@ public static IObservable<DiagnosticListener> AllListeners
         {
             get
             {
+                GC.KeepAlive(HttpHandlerDiagnosticListener.s_instance);
```

Comment:
Definitely put a #if NET45 || NET46 || NETFX around it.  
Also put a comment indicating its purpose (on Desktop platforms to insure HttpHandlerDiagnosticListener.s_instance has been initialized, which in turn will insure that a DiagnosticListener for the HTTP requests is injected into the desktop framework using reflection).  

This hook moved to be in the body of the if (s_allListenerObservable == null)  below (it only has to be called one time).  

 into the framework for HTTP